owncloudDomain/src/main/java/com/owncloud/android/domain/files/usecases/RenameFileUseCase.kt

```diff
@@ -24,18 +26,29 @@ import com.owncloud.android.domain.files.model.OCFile
 import com.owncloud.android.domain.validator.FileNameValidator
 
 class RenameFileUseCase(
-    private val fileRepository: FileRepository
+    private val fileRepository: FileRepository,
+    private val setLastUsageFileUseCase: SetLastUsageFileUseCase,
 ) : BaseUseCaseWithResult<Unit, RenameFileUseCase.Params>() {
 
     private val fileNameValidator = FileNameValidator()
 
     override fun run(params: Params) {
         fileNameValidator.validateOrThrowException(params.newName)
+        val isAvailableLocally = params.ocFileToRename.isAvailableLocally
 
-        return fileRepository.renameFile(
+        fileRepository.renameFile(
             ocFile = params.ocFileToRename,
             newName = params.newName,
         )
+        setLastUsageFile(params.ocFileToRename, isAvailableLocally)
+    }
+
+    private fun setLastUsageFile(file: OCFile, isAvailableLocally: Boolean ){
+        setLastUsageFileUseCase(SetLastUsageFileUseCase.Params(
+            fileId = file.id!!,
```

Comment:
We should not add dose '!!' because it's a source of future crashes.

Reply:
Done